{"version":3,"sources":["gen_inner_err_catch.js"],"names":[],"mappings":"AAAA;;;;;;AAMA,UAAU,MAAV,GAAmB;AACf,QAAI;AACA,cAAM,CAAN;AACH,KAFD,CAEE,OAAO,CAAP,EAAU;AACR,gBAAQ,GAAR,CAAY,MAAZ,EAAoB,CAApB;;AAEH;AACD,UAAM,CAAN;AACH;AACD,IAAI,MAAM,QAAV;AACA,QAAQ,GAAR,CAAY,IAAI,IAAJ,EAAZ;AACA,QAAQ,GAAR,CAAY,IAAI,KAAJ,CAAU,GAAV,CAAZ;AACA,IAAI;AACA,YAAQ,GAAR,CAAY,IAAI,KAAJ,CAAU,GAAV,CAAZ;AACH,CAFD,CAEE,OAAO,CAAP,EAAU;AACR,YAAQ,GAAR,CAAY,MAAZ,EAAoB,CAApB;AACA,YAAQ,GAAR,CAAY,IAAI,IAAJ,EAAZ;AACH","file":"gen_inner_err_catch.js","sourcesContent":["'use strict';\n\n/**\n * 遍历器对象抛错，执行还会从这次yield开始，到下次yield执行结束，不影响遍历\n */\n\nfunction* errCnt() {\n    try {\n        yield 5;\n    } catch (e) {\n        console.log('内部捕获', e);\n        //throw new Error('cry')\n    }\n    yield 6;\n}\nvar itr = errCnt();\nconsole.log(itr.next());\nconsole.log(itr.throw('a'));\ntry {\n    console.log(itr.throw('a'));\n} catch (e) {\n    console.log('外部捕获', e);\n    console.log(itr.next());\n}\n\n//# sourceMappingURL=gen_inner_err_catch.js.map\n\n//# sourceMappingURL=gen_inner_err_catch.js.map"]}